From 10a5c3aa00f3b5643210a53fd83c456bc269300b Mon Sep 17 00:00:00 2001
From: John Greene <jogreene@redhat.com>
Date: Mon, 8 Sep 2014 15:07:59 -0400
Subject: [ethernet] ixgbevf: remove open-coded skb_cow_head

Message-id: <1410188883-6595-45-git-send-email-jogreene@redhat.com>
Patchwork-id: 92365
O-Subject: [RHEL7.1 PATCH 44/48] ixgbevf: remove open-coded skb_cow_head
Bugzilla: 1091124
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Stefan Assmann <sassmann@redhat.com>
RH-Acked-by: John Linville <linville@redhat.com>

BZ: 1091124
Brew: https://brewweb.devel.redhat.com/taskinfo?taskID=7919168
Tested: In beaker. See BZ for details

Signed-off-by: Francois Romieu <romieu@fr.zoreil.com>
Cc: Jesse Brandeburg <jesse.brandeburg@intel.com>
Tested-by: Phil Schmitt <phillip.j.schmitt@intel.com>
Signed-off-by: Jeff Kirsher <jeffrey.t.kirsher@intel.com>
(cherry picked from commit 8f12c0348397ee69af33fc8973b687d83a8759dc)
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
index 96dcb6d..dfa75ff 100644
--- a/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
+++ b/drivers/net/ethernet/intel/ixgbevf/ixgbevf_main.c
@@ -2839,6 +2839,7 @@ static int ixgbevf_tso(struct ixgbevf_ring *tx_ring,
  struct sk_buff *skb = first->skb;
  u32 vlan_macip_lens, type_tucmd;
  u32 mss_l4len_idx, l4len;
+ int err;
 
  if (skb->ip_summed != CHECKSUM_PARTIAL)
   return 0;
@@ -2846,11 +2847,9 @@ static int ixgbevf_tso(struct ixgbevf_ring *tx_ring,
  if (!skb_is_gso(skb))
   return 0;
 
- if (skb_header_cloned(skb)) {
-  int err = pskb_expand_head(skb, 0, 0, GFP_ATOMIC);
-  if (err)
-   return err;
- }
+ err = skb_cow_head(skb, 0);
+ if (err < 0)
+  return err;
 
  /* ADV DTYP TUCMD MKRLOC/ISCSIHEDLEN */
  type_tucmd = IXGBE_ADVTXD_TUCMD_L4T_TCP;
-- 
1.7.1